### DISCUSSION TITLE
Passing data to widgets?

### DISCUSSION THREAD

**Discussion by fholmqvist** [filamentphp/filament#9322] [Help] [answered]
Would it be possible to do something along the lines of:
```
@livewire(MyChart::class, ['ids' => [1, 2, 3, 4]])
```
If so, how?

**Comment by sinnbeck** [comment]
Just like that. You can get it in the mount method of the widget

```
public function mount($ids)
{
    dd($ids);
}

**Reply by fholmqvist** [reply]
I'm afraid not:
```
Method 'App\Filament\Widgets\MyChart::mount()' is not compatible with method 'Filament\Widgets\ChartWidget::mount()'
```

`ChartWidget` accepts no inputs:
```
public function mount(): void
```

Compile time exception:
```
Declaration of App\Filament\Widgets\MyChart::mount($ids) must be compatible with Filament\Widgets\ChartWidget::mount(): void
```

**Reply by sinnbeck** [reply, marked as answer]
Ah I tested with a stats overview widget. It works fine there.

Try adding it as a public property instead

```
public $ids;
```
https://livewire.laravel.com/docs/nesting#passing-props-to-children

**Reply by fholmqvist** [reply]
That worked. Cheers!

**Reply by synsyst** [reply]
I have the issue that I'm trying to get data from an active table filter on a list page into my widget such that the stat widget can be dynamic based on filter in the table on the list page - no luck - any hints would be great. 

Trying the code below gives me the following error:
Filament\Pages\Page::Filament\Pages\{closure}(): Argument #1 ($widget) must be of type Filament\Widgets\WidgetConfiguration|string, array given

ListPage:
`    
public $selectedWeek;


    protected function getHeaderWidgets(): array
    {

        // $selectedWeek = $this->table;
        $selectedWeek = $this->table->getLivewire()->tableFilters['week']['value'];
        // dd($selectedWeek);

        return [
            SurveyMetricsWidget::class => [
                'weekNumber' => $selectedWeek, // Pass the weekNumber to the widget
            ],
        ];
    }
`
    
 
 Widget:   
`
  public $weekNumber;

  protected function getCards(): array
  {
    // dd($this->weekNumber);



    $query = ConsultantSurvey::query();

    // if ($weekFilter) {
    //   $query->where('week', '=', $weekFilter);
    // }

    return [
      Stat::make('Unhappy', $query
        ->where('happiness', '=', 'unhappy')
        ->count()),
      // Add other stats as needed
    ];
  }
`

**Reply by neatstudio** [reply]
> I have the issue that I'm trying to get data from an active table filter on a list page into my widget such that the stat widget can be dynamic based on filter in the table on the list page - no luck - any hints would be great.
> 
> Trying the code below gives me the following error: Filament\Pages\Page::Filament\Pages{closure}(): Argument #1 ($widget) must be of type Filament\Widgets\WidgetConfiguration|string, array given
> 
> ListPage: ` public $selectedWeek;
> 
> ```
> protected function getHeaderWidgets(): array
> {
> 
>     // $selectedWeek = $this->table;
>     $selectedWeek = $this->table->getLivewire()->tableFilters['week']['value'];
>     // dd($selectedWeek);
> 
>     return [
>         SurveyMetricsWidget::class => [
>             'weekNumber' => $selectedWeek, // Pass the weekNumber to the widget
>         ],
>     ];
> }
> ```
> 
> `
> 
> Widget: ` public $weekNumber;
> 
> protected function getCards(): array { // dd($this->weekNumber);
> 
> ```
> $query = ConsultantSurvey::query();
> 
> // if ($weekFilter) {
> //   $query->where('week', '=', $weekFilter);
> // }
> 
> return [
>   Stat::make('Unhappy', $query
>     ->where('happiness', '=', 'unhappy')
>     ->count()),
>   // Add other stats as needed
> ];
> ```
> 
> } `

Hey man, where you saw this `Filament\Widgets\WidgetConfiguration|string`，you must set a value like `Filament\Widgets\WidgetConfiguration`

so the code like this:
```php
protected function getHeaderWidgets(): array
{
     $selectedWeek = $this->table->getLivewire()->tableFilters['week']['value'];
     // dd($selectedWeek);
 
     return [
         new \Filament\Widgets\WidgetConfiguration(SurveyMetricsWidget::class, [
             'weekNumber' => $selectedWeek, // Pass the weekNumber to the widget
         ]),
         // SurveyMetricsWidget::class => [
         //    'weekNumber' => $selectedWeek, // Pass the weekNumber to the widget
         // ],
     ];
 }
 ```
